components/builder-api-client/src/builder.rs

```diff
@@ -462,6 +462,35 @@ impl BuilderAPIProvider for BuilderAPIClient {
                       progress)
     }
 
+    /// Create an origin
+    ///
+    ///  # Failures
+    ///
+    ///  * Remote builder is not available
+    ///  * Origin already exists
+    ///  * Y'aint got no auth
+    fn create_origin(&self, origin: &str, token: &str) -> Result<()> {
+        let path = format!("depot/origins");
```

Comment:
Since you're only ever using this as a `&str` in the call to `self.0.post()` below, you shouldn't need this line at all.  You should be able to just pass `"depot/origins"` to the call to `self.0.post()` directly.